lib/kinesis_client.rb

```diff
@@ -84,19 +78,31 @@ def push_batch(batch)
 
     return_message = {
       failures: resp.failed_record_count,
-      error_messages: resp.records.map {|record| record.error_message }.compact
+      failures_data: filter_failures(resp),
+      # error_messages: resp.records.map { |record| record.error_message }.compact
     }
 
     $logger.info("Message sent to #{config[:stream_name]} #{return_message}") if $logger
 
-    return {
+    {
       "code": "200",
       "message": return_message.to_json
     }
   end
 
   def push_records
-    @batch.each_slice(@batch_size) {|slice| push_batch slice}
-    @batch = []
+    @records.each_slice(@batch_size) { |slice| push_batch(slice) } if @records.length > 0 
```

Comment:
OK so we didn't talk about this, but it looks like the way this currently works it is basically swallowing the error messages, which we don't want. We should at least log the response of `push_batch`

See below where I have a note about this in the context of testing.